### DISCUSSION TITLE
[Session Key validation]: unlock context variables access in the transaction validation hook of Smart Accounts

### DISCUSSION THREAD

**Discussion by Haypierre** [zkSync-Community-Hub/zksync-developers#676] [Feature requests]
### Network

ZKsync Era

### Provide a brief description of the feature you would like to see implemented.

Today, context variables (e.g., `block.timestamp`) are prohibited in account logic when verifying a transaction.

It's an issue for session key validation - the only workaround is to [move the logic in the execution hook](https://github.com/openfort-xyz/openfort-zksync-contracts/blob/f5c0e1c99de945bfd034dfd5ec09dba00c9ef194/contracts/core/base/BaseOpenfortAccount.sol#L270) which is suboptimal:

* if the user proceeds with an invalid session key it will fail and consume gas
* recover the signature twice (in the verification hook AND the execution hook)

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

Users AND developers

**Comment by bxpana** [comment]
Thanks for this feature request! If you know others that would like to show their support and upvote this send it to them so we know how to prioritize this for the community.
